include/hal/common-mock-func.sh

```diff
@@ -0,0 +1,486 @@
+
+  echo "${FUNCNAME[0]}: using ${_tool}..."
+
+  return 0
+}
+
+################################################################################
+# flashrom
+################################################################################
+TEST_FLASH_LOCK="${TEST_FLASH_LOCK:-}"
+TEST_BOARD_HAS_FD_REGION="${TEST_BOARD_HAS_FD_REGION:-true}"
+TEST_BOARD_FD_REGION_RW="${TEST_BOARD_FD_REGION_RW:-true}"
+TEST_BOARD_HAS_ME_REGION="${TEST_BOARD_HAS_ME_REGION:-true}"
+TEST_BOARD_ME_REGION_RW="${TEST_BOARD_ME_REGION_RW:-true}"
+TEST_BOARD_ME_REGION_LOCKED="${TEST_BOARD_ME_REGION_LOCKED:-}"
+TEST_BOARD_HAS_GBE_REGION="${TEST_BOARD_HAS_GBE_REGION:-true}"
+TEST_BOARD_GBE_REGION_RW="${TEST_BOARD_GBE_REGION_RW:-true}"
+TEST_BOARD_GBE_REGION_LOCKED="${TEST_BOARD_GBE_REGION_LOCKED:-}"
+TEST_COMPATIBLE_EC_VERSINO="${TEST_COMPATIBLE_EC_VERSINO:-}"
+# This is 2*1024*1024:
+TEST_FLASH_CHIP_SIZE="${TEST_FLASH_CHIP_SIZE:-2097152}"
+
+flashrom_check_flash_lock_mock(){
+# For flash lock testing, for more inf. check check_flash_lock func.:
+  if [ "$TEST_FLASH_LOCK" = "true" ]; then
+    echo "PR0: Warning:.TEST is read-only\|SMM protection is enabled" > /tmp/check_flash_lock.err
+    return 1
+  fi
+
+  return 0
+}
+
+flashrom_flash_chip_name_mock(){
+# For flash chip name check emulation, for more inf. check check_flash_chip
+# func.:
+    echo "Test Flash Chip" 1>&1
+
+    return 0
+}
+
+flashrom_flash_chip_size_mock(){
+# For flash chip size check emulation, for more inf. check check_flash_chip
+# func..
+  echo "$TEST_FLASH_CHIP_SIZE" 1>&1
+
+  return 0
+}
+
+flashrom_check_intel_regions_mock(){
+# For flash regions check emulation, for more inf. check check_intel_regions
+# func.:
+  [ "$TEST_BOARD_HAS_FD_REGION" = "true" ] && echo "Flash Descriptor region" 1>&1
+  [ "$TEST_BOARD_FD_REGION_RW" = "true" ] && echo "Flash Descriptor region.testread-write" 1>&1
+
+  [ "$TEST_BOARD_HAS_ME_REGION" = "true" ] && echo "Management Engine region" 1>&1
+  [ "$TEST_BOARD_ME_REGION_RW" = "true" ] && echo "Management Engine region.testread-write" 1>&1
+  [ "$TEST_BOARD_ME_REGION_LOCKED" = "true" ] && echo "Management Engine region.testlocked" 1>&1
+
+  [ "$TEST_BOARD_HAS_GBE_REGION" = "true" ] && echo "Gigabit Ethernet region" 1>&1
+  [ "$TEST_BOARD_GBE_REGION_RW" = "true" ] && echo "Gigabit Ethernet region.testread-write" 1>&1
+  [ "$TEST_BOARD_GBE_REGION_LOCKED" = "true" ] && echo "Gigabit Ethernet region.testlocked" 1>&1
+
+  return 0
+}
+
+flashrom_read_flash_layout_mock(){
+# For checking flash layout for further flashrom arguments selection, for more
+# inf. check set_flashrom_update_params function.
+#
+# TODO: this one can be deleted in future and replaced with read_firm_mock,
+# which will create a binary with needed bytes appropriately set.
+  # For -r check flashrom man page:
+  local _file_to_write_into
+  # It won't work with "$*", so make shellcheck ignore it:
+  # shellcheck disable=SC2048
+  _file_to_write_into=$(parse_for_arg_return_next "-r" $*)
```

Comment:
"$*" passes all parameters as a single argument, "$@" passes all parameters as separate arguments (without separating parameters of type e.g. `path to file with spaces.txt` into separate arguments).
Without quotes I think it works the same.

You should probably use "$@" everywhere or almost everywhere instead of $*

Reply:
Addressed [here](https://github.com/Dasharo/dts-scripts/compare/8e94cbc7f99ffe845b06f6f742c9b9863dec795e..992062280bbd17d42f45e14d98c3e682971d4ee4)